### PR TITLE
Gpeacock/sign work

### PR DESCRIPTION
**Pull request by gpeacock**

### CODE REVIEW

File: src/stream.rs [hunk 1]
```diff
@@ -66,34 +67,75 @@ pub trait Stream: Send + Sync {
     fn write_stream(&self, data: Vec<u8>) -> StreamResult<u64>;
 }
 
-impl Read for dyn Stream {
+impl Stream for Box<dyn Stream> {
```
Comment (scouten-adobe) [inline src/stream.rs:70]: OMG I had no idea you could do such a thing!

File: tests/c/c2pa.h [hunk 2]
```diff
@@ -30,12 +30,62 @@
...
+ * # Example
+ * ```
+ * use c2pa::SignerConfig;
+ * let config = SignerConfig {
+ *    alg: "Rs256".to_string(),
+ *    certs: vec![vec![0; 10]],
+ *    time_authority_url: Some("http://example.com".to_string()),
+ *    use_ocsp: true,
+ * };
+ */
```
Comment (scouten-adobe) [inline tests/c/c2pa.h:69]: Example needs closing ```

**Review by scouten-adobe — CHANGES_REQUESTED** [review]
There's a lot of hard work in here. Really nice to see it coming along. A few comments and questions, but overall really solid!

File: src/c2pa.udl [hunk 3]
```diff
@@ -42,7 +43,7 @@ interface ManifestStoreReader {
   constructor();
 
   [Throws=StreamError]
-  string read([ByRef] string format, Stream reader);
+  string read_stream([ByRef] string format, [ByRef] Stream reader);
```
[the remark this answers is not in the document]
Comment (gpeacock) [reply src/c2pa.udl:46]: _mut is a Rust specific naming convention. The UDL file generates bindings for many languages where I don't think the _mut conventions would make any sense.

File: src/manifest_builder.rs [hunk 4]
```diff
@@ -59,30 +78,102 @@ impl ManifestBuilder {
...
-            //             Err(_) => {
-            //                 None
-            //             }
-            //         })
+        let mut input = StreamAdapter::from(input_mut);
+        let mut output = StreamAdapter::from(output_mut);
+        self.read(signer, &mut input, &mut output)
+    }
+
+    pub fn read(
```
[the remark this answers is not in the document]
Comment (gpeacock) [reply src/manifest_builder.rs:96]: This was totally a mistake while doing a function rename, thanks. I'm changing it back to "sign"

**Pull request merged**